R/board_connect_url.R

```diff
@@ -0,0 +1,155 @@
+#' Use a vector of Posit Connect vanity URLs as a board
+#'
+#' @description
+#' `board_connect_url()` lets you build up a board from individual
+#' [vanity urls](https://docs.posit.co/connect/user/content-settings/#custom-url).
+#' To use a Posit Connect board like this one or [board_connect()], you need to
+#' first authenticate. The easiest way to do so is by launching **Tools** -
+#' **Global Options** - **Publishing** - **Connect**, and follow the
+#' instructions.
+#'
+#' `board_connect_url()` is read only and does not support the use of a
+#' [manifest file][write_board_manifest()].
+#'
+#' @param vanity_urls A named character vector of
+#'   [Connect vanity URLs](https://docs.posit.co/connect/user/content-settings/#custom-url).
+#'   This board is read only, and the best way to write to a pin on Connect is
+#'   [board_connect()].
+#' @family boards
+#' @inheritParams new_board
+#' @inheritParams board_connect
+#' @export
+#' @examplesIf interactive() || identical(Sys.getenv("IN_PKGDOWN"), "true")
+#' board <- board_connect_url(c(
+#'     my_vanity_url_pin = "https://colorado.posit.co/rsc/great-numbers/"
+#' ))
+#'
+#' board %>% pin_read("my_vanity_url_pin")
+#'
+board_connect_url <- function(vanity_urls,
+                              auth = c("auto", "manual", "envvar", "rsconnect"),
+                              server = NULL,
+                              account = NULL,
+                              key = NULL,
+                              cache = NULL,
+                              use_cache_on_failure = is_interactive()) {
+
+  server <- rsc_server(auth, server, account, key)
+  cache <- cache %||% board_cache_path(paste0("connect-url-", hash(server$url)))
+
+  board <- new_board_v1(
+    "pins_board_connect_url",
+    vanity_urls = vanity_urls,
```

Comment:
I am calling this `vanity_urls` for now until I figure out if we need the `url` element in the board still. Maybe we end up with just plain `urls`, in which case, I can reuse some of the `board_url()` methods.

Reply:
Turns out I'm not going to use `urls` because I need the `url` element on the board to use `rsc_GET()` and similar. So let's stick with `vanity_urls` to distinguish it.